x/leverage/keeper/grpc_query.go

```diff
@@ -62,6 +62,35 @@ func (q Querier) RegisteredTokens(
 	}, nil
 }
 
+func (q Querier) RegisteredTokenMarkets(
+	goCtx context.Context,
+	req *types.QueryRegisteredTokenMarkets,
+) (*types.QueryRegisteredTokenMarketsResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	tokens := q.Keeper.GetAllRegisteredTokens(ctx)
+	markets := []types.TokenMarket{}
+
+	for _, token := range tokens {
+		marketSumnmary, err := q.MarketSummary(goCtx, &types.QueryMarketSummary{Denom: token.BaseDenom})
+		if err != nil {
+			// absorb overall query error into struct, which may be empty, but proceed with this query
+			marketSumnmary.Errors += err.Error()
```

Comment:
why do we return errors as a String as a part of the "normal" object response?

Reply:
Mimics `MarketSummary` behavior.

This was before `errors.Join` existed, and it runs multiple calculations where we need to return the fields we can even if some of the others error. The string field kept the API simple (always a known, readable type for debugging)